### ISSUE TITLE
Posts of different types with same slug cause collisions during queries

### ISSUE THREAD

**Issue by ravewebdev** [wp-graphql/wp-graphql#1695]
## Expected behavior ##

If there are multiple posts of different types with the same slug, a standard GraphQL query should be able to differentiate between them and return the correct post based on the specified post type.

E.g.,

A post named "Test" has the slug "test" and should be returned by the following query:

```
query getPost {
  post(id: "test", idType: SLUG) {
    title
  }
}
```

A page named "Test" has the slug "test" and should be returned by the following query:

```
query getPage {
  page(id: "test", idType: URI) {
    title
  }
}
```

An event (CPT) named "Test" has the slug "test" and should be returned by the following query:

```
query getEvent {
  event(id: "test", idType: SLUG) {
    title
  }
}
```

## Actual behavior ##

The queries above do not behave as expected. They seem to query for the first post, of any type, matching that slug, rather than restricting the query to the specified post type. For example, the `getPage` query above on my local environment is returning the event post instead of the page.

Running WP GraphQL v1.1.3

**Comment by colbyfayock** [comment]
I think I'm seeing something similar here:

![image](https://user-images.githubusercontent.com/1045274/124362435-c9e1dd00-dc02-11eb-937b-38b30bc01e58.png)

I'm going to just change my slug for now for the one that's conflicting

**Comment by jasonbahl** [comment]
I believe this has been resolved by #1994 which was released as part of v1.5.2 today. 

I'm going to close this, but if you're able to reproduce after updating to ^=v1.5.2 please open a new issue and reference this one for context.
